streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1248,15 +1248,16 @@ protected StreamTask createStreamTask(final TaskId id, final Collection<TopicPar
     }
 
     private Producer<byte[], byte[]> createProducer(final TaskId id) {
-        final Map<String, Object> producerConfigs = config.getProducerConfigs(threadClientId);
```

Comment:
UUID will only be used if user does not specify the `clientId`. If a client id is specified then the metric prefix would be `thread.[clientId]-streamThread-[sequence-number]` right?